src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit } from '@angular/core';
+
+import { NodeEvent, TreeModel } from 'ng2-tree';
+
+import { DashboardService } from '../../../shared/api/dashboard.service';
+
+@Component({
+  selector: 'cd-crushmap',
+  templateUrl: './crushmap.component.html',
+  styleUrls: ['./crushmap.component.scss']
+})
+export class CrushmapComponent implements OnInit {
+  panelTitle: string;
+  tree: TreeModel;
+  metadata: any;
+  metadataKeyMap: { [key: number]: number } = {};
+
+  constructor(private dashboardService: DashboardService) {
+    this.panelTitle = 'Cluster Crush Map';
+  }
+
+  ngOnInit() {
+    this.dashboardService.getHealth().subscribe((data: any) => {
+      this.tree = this._abstractTreeData(data);
+    });
+  }
+
+  _abstractTreeData(data: any): TreeModel {
```

Comment:
Could you please add some tests that verify the correct tree creation?

Reply:
Agreed, please make sure to include some tests.

Reply:
@Devp00l A test case for correct tree was added.